Observer/AssignDataObserver.php

```diff
@@ -32,6 +32,8 @@ public function execute(Observer $observer)
         }
         if (array_key_exists('type', $additionalData)) {
             $paymentInfo->setAdditionalInformation('type', $additionalData['type']);
+        } else {
```

Comment:
If you're always setting this value, then you don't need the `if` - you could just do:

```
$paymentInfo->setAdditionInformation('type', $additionalData['type'] ?? null);
```